include/TTipHit.h

```diff
@@ -27,18 +28,42 @@ class TTipHit : public TGRSIDetectorHit {
     Double_t slow_amplitude;
     Double_t gamma_amplitude;
    
-    Double_t fit_time;
+	Int_t	 tip_channel;
+
+    Double_t    time_fit;
+    Double_t    sig2noise;
 
   public:
     /////////////////////////    /////////////////////////////////////
-    inline void SetFilterPattern(const int &x)   { filter   = x; }   //! 
+    inline void SetFilterPattern(const int &x)    { filter   = x; }   //! 
     inline void SetPID(Double_t x)                { fPID = x;     }   //!
+	inline void SetTipChannel(const int x)		  { tip_channel = x; } //!
 
     inline Int_t    GetFiterPatter()           { return filter;   }  //!
     inline Double_t GetPID()                   { return fPID;      }  //!
+	inline Double_t GetFitTime()			   { return time_fit;	} //!
+	inline Double_t GetSignalToNoise()		   { return sig2noise;	} //!
+	inline Int_t	GetTipChannel()			   { return tip_channel; } //!
 
     bool   InFilter(Int_t);                                         //!
 
+    void SetVariables(TFragment &frag) { SetCfd(frag.GetCfd());
+                                         SetCharge(frag.GetCharge());
+                                         SetTimeStamp(frag.GetTimeStamp()); }
+
+	void SetUpNumbering(TChannel &chan) { MNEMONIC mnemonic;
+										  TChannel *channel = GetChannel();
+										  if(!channel){
+										    Error("SetDetector","No TChannel exists for address %u",GetAddress());
+										    return;
+										  }
+										  ClearMNEMONIC(&mnemonic);
+										  ParseMNEMONIC(channel->GetChannelName(),&mnemonic); 
+										  Int_t tmp = (int16_t)atoi(mnemonic.arraysubposition.c_str()); 
+										  SetTipChannel((Int_t)(10*mnemonic.arrayposition + tmp)); }
+
```

Comment:
You gain the fact that you aren’t writing the detector and sub detector to disk. You can get this information from the TChannel/mnemonic itself, and you can calculate it faster than reading it from disk. You can leave the tip_channel in there the way you are doing it. I would just at the very least make it transient and then do similar checks that we are doing for the other transient members.

> On Oct 28, 2015, at 11:43 AM, jhenderson88 <notifications@github.com> wrote:
> 
> In include/TTipHit.h <https://github.com/GRIFFINCollaboration/GRSISort/pull/554#discussion_r43291583>:
> 
> >  
> >      bool   InFilter(Int_t);                                         //!
> >  
> > +    void SetVariables(TFragment &frag) { SetCfd(frag.GetCfd());
> > +                                         SetCharge(frag.GetCharge());
> > +                                         SetTimeStamp(frag.GetTimeStamp()); }
> > +
> > +	void SetUpNumbering(TChannel &chan) { MNEMONIC mnemonic;
> > +										  TChannel *channel = GetChannel();
> > +										  if(!channel){
> > +										    Error("SetDetector","No TChannel exists for address %u",GetAddress());
> > +										    return;
> > +										  }
> > +										  ClearMNEMONIC(&mnemonic);
> > +										  ParseMNEMONIC(channel->GetChannelName(),&mnemonic); 
> > +										  Int_t tmp = (int16_t)atoi(mnemonic.arraysubposition.c_str()); 
> > +										  SetTipChannel((Int_t)(10*mnemonic.arrayposition + tmp)); }
> > +
> I can do that, although I'm not entirely sure what we gain from doing so? We're storing two values (one in TGRSIDetectorHit, one in TTipHit) one way the other.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/GRIFFINCollaboration/GRSISort/pull/554/files#r43291583>.
>